packages/tpetra/core/test/CrsGraph/CrsGraph_UnitTests_Swap.cpp

```diff
@@ -125,28 +125,25 @@ TEUCHOS_STATIC_SETUP()
 // gbl_num_columns: Max # of columns in the matrix-representation of the graph.
 //                  This should be >= the highest value of v from all edges (u,v) in edges.
 //                  Note: u and v are 0-indexed, so if the highest v is 11, then this should be 12.
-template<class LO, class GO, class Node, class comm_type>
-Teuchos::RCP<Tpetra::CrsGraph<LO, GO, Node>>
-generate_crsgraph(Teuchos::RCP<comm_type>                      comm,
-                  const std::vector<std::pair<GO, GO>>&     gbl_edges,
-                  const std::vector<std::pair<GO, size_t>>& gbl_row_owners,
-                  const size_t                              gbl_num_columns,
-                  const bool                                do_fillComplete=true)
+template<class LO, class GO, class Node>
+Teuchos::RCP<Tpetra::CrsGraph<LO, GO, Node> >
+generate_crsgraph(Teuchos::RCP<Teuchos::Comm<int> >&      comm,
+                  const std::vector<std::pair<GO, GO> >&  gbl_edges,
+                  const std::vector<std::pair<GO, int> >& gbl_row_owners,
+                  const size_t                            gbl_num_columns,
+                  const bool                              do_fillComplete=true)
 {
     using Teuchos::Comm;
 
-    typedef Tpetra::CrsGraph<LO, GO, Node> graph_type;
-    typedef Tpetra::Map<LO, GO, Node>      map_type;
-
-    typedef typename std::map<GO, size_t> map_rows_type;      // row_id, num_entries
-
-
-    typedef typename std::vector<GO>        vec_go_type;
-    typedef typename std::map<GO, vec_go_type> map_row_to_cols_type;
+    using graph_type           = Tpetra::CrsGraph<LO,GO,Node>;      // Tpetra CrsGraph type
+    using map_type             = Tpetra::Map<LO,GO,Node>;           // Tpetra Map type
+    using map_rows_type        = std::map<GO,int>;                  // map rows to pid's
+    using vec_go_type          = std::vector<GO>;                   // vector of GlobalOrdinals
+    using map_row_to_cols_type = std::map<GO, vec_go_type>;         // Map rows to columns
 
     const bool verbose = Tpetra::Details::Behavior::verbose();
 
-    const size_t comm_rank = comm->getRank();
+    const int comm_rank = (size_t)comm->getRank();
```

Comment:
Don't cast to `size_t`.  It's not a `size_t` and it never was :-)

Reply:
gah, missed that when I was changing everything to int.  Will fix and push the update :p

Reply:
BTW, thanks for looking at this... I greatly appreciate your attention to detail.  Its so easy to miss a thing like this when the compiler doesn't give me any warnings on it locally and my eyes just don't want to see the miscast since that was something I tossed in going down the path of using `size_t` before I just fixed my use of `comm_rank` overall from `size_t` to `int`. :)

Also, I'm sure the autotester application is hating on me right now, I keep letting it kick off my jobs then it keeps having to kill the tests b/c I'm updating my PR. haha.  Normally that's a WIP thing, but I keep thinking 'oh, this is a minor fix' so it's totes good. :p Anyway, this one is the last tweak I think  <knock on wood>.  The build cluster for the tests isn't heavily loaded today so it's not a big deal so its good

Reply:
@william76 I'm being a bit of a pain about "attention to detail" but it's so much easier to read the code and be sure that it's correct when it has a consistent style, so thanks :-D